tests/test_callprofiler.py

```diff
@@ -0,0 +1,106 @@
+# Copyright (c) 2024, Quentin Tsai <quentin.tsai.tw@gmail.com>
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# - Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+# - Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+# - Neither the name of the copyright holder nor the names of its contributors
+#   may be used to endorse or promote products derived from this software
+#   without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+import unittest
+import time
+import re
+import os
+from functools import wraps
+import modmesh
+
+
+# https://docs.python.org/3/library/functools.html#functools.wraps
+def profile_function(func):
```

Comment:
This will be a helpful utility function for developing modmesh code, e.g., GUI (FYI @j8xixo12).  Eventually I think a more tailored implementation should be made available somewhere under the `modmesh` Python package.

At the time being, could you please add a comment for the above-mentioned argument?

When it comes to the to-be-useful-and-mature helper function in the future, I don't like the function name (`profile_function`).  But it is OK for a local helper in the test module.

Reply:
I am not seeing the descriptions in comment.  Please point it out.

Reply:
> This will be a helpful utility function for developing modmesh code, e.g., GUI (FYI @j8xixo12). Eventually I think a more tailored implementation should be made available somewhere under the `modmesh` Python package.
> 
> At the time being, could you please add a comment for the above-mentioned argument?

By "add a comment for ...", I meant to add the comment in the code, so that a future maintainer will read it and perform the refactoring.